### PR TITLE
Feature/budget enhance print css

### PR DESCRIPTION
**Pull request by mromeroval**
Issue:
https://trello.com/c/YPnoXCGQ

### CODE REVIEW

File: app/budget/directives/budgetScenarioToolbar/budgetScenarioToolbar.js [hunk 1]
```diff
@@ -72,6 +75,11 @@ let budgetScenarioToolbar = function($window, $location, $routeParams, $rootScop
 				scope.state.selectedBudgetScenario.activeTermsBlob = setCharAt(scope.state.selectedBudgetScenario.activeTermsBlob, index, newValue);
 				BudgetActions.updateBudgetScenario(scope.state.selectedBudgetScenario);
 			};
+
+			scope.print = function() {
+				window.print();		
+			};
+
```
Comment (ltwheeler) [inline app/budget/directives/budgetScenarioToolbar/budgetScenarioToolbar.js]: Remove extra space

File: app/budget/directives/budgetCosts/budgetCostsPrint/budgetCostsPrint.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+				'05': 'Summer Session 1',
+				'06': 'Summer Special Session',
+				'07': 'Summer Session 2',
+				'08': 'Summer Quarter',
+				'09': 'Fall Semester',
+				'10': 'Fall Quarter',
+				'01': 'Winter Quarter',
+				'02': 'Spring Semester',
+				'03': 'Spring Quarter'
+			};
```
[the remark this answers is not in the document]
Comment (mromeroval) [reply app/budget/directives/budgetCosts/budgetCostsPrint/budgetCostsPrint.js]: I was actually using TermService  :)

File: app/budget/budget.css [hunk 3]
```diff
@@ -130,11 +130,16 @@
...
+	}
 
-  .do-not-print {
+	.do-not-print {
 		display: none;
-  }
+	}
+	
+	.do-print{
+		display: block !important;
```
Comment (ltwheeler) [inline app/budget/budget.css]: The 'important' flag should only be used in extreme situations, as they make our CSS even harder to predict. The rare times I do use them its usually related to overriding something from bootstrap that used the important flag first.

Does this not work without the important flag?

File: app/budget/budget.css [hunk 4]
```diff
@@ -130,11 +130,16 @@
...
+	}
+
```
Comment (ltwheeler) [inline app/budget/budget.css]: Extra space here

File: app/budget/budget.html [hunk 5]
```diff
@@ -35,7 +35,9 @@
 
 		<div id="toast-container" class="toast-top-right" ng-if="$root.toast.message.length > 0">
 			<div class="toast toast-error" style="display: block;">
+				HERE
```
Comment (ltwheeler) [inline app/budget/budget.html]: Leftover debug message?

File: app/budget/budget.html [hunk 6]
```diff
@@ -35,7 +35,9 @@
...
 				<div class="toast-message" ng-bind="$root.toast.message">
+						HERE
```
Comment (ltwheeler) [inline app/budget/budget.html]: Leftover debug message?

File: app/budget/budgetCtrl.js [hunk 7]
```diff
@@ -25,6 +25,7 @@ class BudgetCtrl {
 	initialize () {
 		var _self = this;
 
+
```
Comment (ltwheeler) [inline app/budget/budgetCtrl.js]: extra spaces here

File: app/budget/directives/budgetCosts/budgetCosts.html [hunk 8]
```diff
@@ -135,3 +135,5 @@
 		</div>
 	</ipa-tabs>
 </div>
+
+
```
Comment (ltwheeler) [inline app/budget/directives/budgetCosts/budgetCosts.html]: Extra spaces

File: app/budget/directives/budgetCosts/budgetCostsPrint/budgetCostsPrint.css [hunk 9]
```diff
@@ -0,0 +1,223 @@
+
+.do-print{
+	display: none;
+}
+
+@media print {
+  @page {
```
Comment (ltwheeler) [inline app/budget/directives/budgetCosts/budgetCostsPrint/budgetCostsPrint.css]: Indentation seems to be flipping between 2 spaces and a tab

File: app/budget/directives/budgetCosts/budgetCostsPrint/budgetCostsPrint.css [hunk 10]
```diff
@@ -0,0 +1,223 @@
...
+		padding: 5px;
+		color: #b1b1b1;
+		text-align: center;
+	}
+	
+
+/* SCHEDULE COST STYLE */
+
+	.budget-costs-print, .budget-costs-print__table{
+		width: 670px !important;
```
Comment (ltwheeler) [inline app/budget/directives/budgetCosts/budgetCostsPrint/budgetCostsPrint.css]: Only use the important flag as a last resort, can this be re-written without the flag?

File: app/budget/directives/budgetCosts/budgetCostsPrint/budgetCostsPrint.css [hunk 11]
```diff
@@ -0,0 +1,223 @@
...
+	}
+
+	.budget-costs-print__table {
+		border-collapse: collapse;
+		font: 7.5pt "Helvetica Neue", Helvetica, "Noto Sans", sans-serif;
+	}
+	
+	.budget-costs-print__table-page-brake {
+		page-break-before: auto;
+		page-break-inside: avoid !important;
```
Comment (ltwheeler) [inline app/budget/directives/budgetCosts/budgetCostsPrint/budgetCostsPrint.css]: Only use the important flag as a last resort, can this be re-written without the flag?

File: app/budget/directives/budgetCosts/budgetCostsPrint/budgetCostsPrint.css [hunk 12]
```diff
@@ -0,0 +1,223 @@
...
+		border: 1px solid #e3e3e3;
+		text-align: center;
+	}
+
+	.budget-costs-print__table > tbody > tr > td {
+		border-top:0;
+	}
+
+	.budget-costs-print__table > tbody > tr > td {
+		padding: 0px !important;
```
Comment (ltwheeler) [inline app/budget/directives/budgetCosts/budgetCostsPrint/budgetCostsPrint.css]: Only use the important flag as a last resort, can this be re-written without the flag?

File: app/budget/directives/budgetCosts/budgetCostsPrint/budgetCostsPrint.css [hunk 13]
```diff
@@ -0,0 +1,223 @@
...
+		font-family: inherit;
+		font-weight: 500;
+		border-bottom: 1px solid #4C4C4C;
+	}
+
+	caption {
+		font-size: 9pt;
+		font-weight: bold;
+		padding-bottom: 2px;
+		color: #333333 !important;
```
Comment (ltwheeler) [inline app/budget/directives/budgetCosts/budgetCostsPrint/budgetCostsPrint.css]: Only use the important flag as a last resort, can this be re-written without the flag?

File: app/budget/directives/budgetCosts/budgetCostsPrint/budgetCostsPrint.js [hunk 14]
```diff
@@ -0,0 +1,57 @@
...
+			instructorAssignmentOptions: '<',
+			regularInstructorAssignmentOptions: '<',
+			isLiveDataScenario: '<',
+			lineItems: '<'
+		},
+		link: function (scope){
+			var json = JSON.parse(localStorage.workgroup);
+			scope.workgroupName = json.name;
+
+			scope.allTerms = {
```
Comment (ltwheeler) [inline app/budget/directives/budgetCosts/budgetCostsPrint/budgetCostsPrint.js]: This should not be hardcoded in the directive, use a helper service to get these values (I think term service may already have this available? unsure).

File: app/budget/directives/budgetSummary/budgetSummary.css [hunk 15]
```diff
@@ -71,4 +71,4 @@
   padding: 5px;
   color: #b1b1b1;
 	text-align: center;
-}
+}
```
Comment (ltwheeler) [inline app/budget/directives/budgetSummary/budgetSummary.css]: missing new line

File: app/budget/directives/budgetCosts/budgetCostsPrint/budgetCostsPrint.js (hunk 14 above)
Comment (mromeroval) [reply app/budget/directives/budgetCosts/budgetCostsPrint/budgetCostsPrint.js]: I was already using TermService. Just forgot to delete that variable :)

File: app/budget/directives/budgetCosts/budgetCostsPrint/budgetCostsPrint.css (hunk 13 above)
Comment (mromeroval) [reply app/budget/directives/budgetCosts/budgetCostsPrint/budgetCostsPrint.css]: No need of !important and not an important color either

File: app/budget/budget.css (hunk 3 above)
Comment (mromeroval) [reply app/budget/budget.css]: No need of !important in any of the cases pointed out.

File: app/budget/directives/budgetSummary/budgetSummary.css (hunk 15 above)
Comment (mromeroval) [reply app/budget/directives/budgetSummary/budgetSummary.css]: Fixed indentation in line 73  too

File: app/budget/directives/budgetCosts/budgetCostsPrint/budgetCostsPrint.css [hunk 16]
```diff
@@ -0,0 +1,223 @@
...
+
+	/* FUNDS STYLE */
+
+	.line-items-print__table {
+		width: 670px;
+		margin-top: 20px;
+		margin-bottom: 20px;
+	}
+
+	.line-items-print__table > thead > tr > th,
```
[the remark this answers is not in the document]
Comment (mromeroval) [reply app/budget/directives/budgetCosts/budgetCostsPrint/budgetCostsPrint.css]: I think I've got it. Thanks!

File: app/budget/directives/budgetScenarioToolbar/budgetScenarioDropdown/budgetScenarioDropdown.css [hunk 17]
```diff
@@ -32,3 +32,21 @@
...
 	justify-content: space-between;
 }
+
+@media print {
+	@page {
+		margin: 1cm;
+	}
+
+	.do-not-print {
+			display: none !important;
```
Comment (ltwheeler) [inline app/budget/directives/budgetScenarioToolbar/budgetScenarioDropdown/budgetScenarioDropdown.css]: Indentation isn't correct here

Comment (ltwheeler) [inline app/budget/directives/budgetScenarioToolbar/budgetScenarioDropdown/budgetScenarioDropdown.css]: Is the important flag absolutely necessary? And if so, why is that?

Using the important flag quickly complicates CSS and makes future work harder. It should be used as a last resort.

File: app/budget/directives/budgetScenarioToolbar/budgetScenarioToolbar.js [hunk 18]
```diff
@@ -7,14 +7,17 @@ let budgetScenarioToolbar = function($window, $location, $routeParams, $rootScop
 		restrict: 'E', // Use this via an element selector <budget-scenario-dropdown></budget-scenario-dropdown>
 		template: require('./budgetScenarioToolbar.html'), // directive html found here:
 		replace: true, // Replace with the template
+		
 		scope: {
-			state: '<'
+			state: '<',
+			printBudget: '&',
```
Comment (ltwheeler) [inline app/budget/directives/budgetScenarioToolbar/budgetScenarioToolbar.js]: This scope variable appears to not be used, am I missing something?

File: app/budget/directives/budgetScenarioToolbar/budgetScenarioToolbar.js [hunk 19]
```diff
@@ -7,14 +7,17 @@ let budgetScenarioToolbar = function($window, $location, $routeParams, $rootScop
...
 		},
 		link: function (scope) {
 			scope.displayScenarioRenameUI = false;
 			scope.newScenarioName = angular.copy(scope.state.selectedBudgetScenario.name); // eslint-disable-line no-undef
 			scope.isNewScenarioNameValid = true;
 			scope.validationError = "";
+			scope.printingMode = false;
```
Comment (ltwheeler) [inline app/budget/directives/budgetScenarioToolbar/budgetScenarioToolbar.js]: This scope variable appears to not be used, am I missing something?

File: app/budget/directives/budgetScenarioToolbar/budgetScenarioDropdown/budgetScenarioDropdown.css (hunk 17 above)
Comment (mromeroval) [reply app/budget/directives/budgetScenarioToolbar/budgetScenarioDropdown/budgetScenarioDropdown.css]: No need of !important flag

File: app/budget/directives/budgetScenarioToolbar/budgetScenarioToolbar.js (hunk 18 above)
Comment (mromeroval) [reply app/budget/directives/budgetScenarioToolbar/budgetScenarioToolbar.js]: Old code, not using it anymore

File: app/budget/directives/budgetScenarioToolbar/budgetScenarioToolbar.js (hunk 19 above)
Comment (mromeroval) [reply app/budget/directives/budgetScenarioToolbar/budgetScenarioToolbar.js]: Old code, not using it anymore

**Pull request merged**